src/ucp/core/ucp_mm.c

```diff
@@ -843,6 +841,29 @@ static ucs_status_t ucp_memh_init_uct_reg(ucp_context_h context, ucp_mem_h memh,
     return status;
 }
 
+static ucs_status_t
+ucp_memh_init_all_uct_reg(ucp_context_h context, ucp_mem_h memh,
+                          unsigned uct_flags, const char *alloc_name)
+{
+    ucs_memory_type_t mem_type = memh->mem_type;
+    ucp_md_map_t reg_md_map    = context->reg_md_map[mem_type];
+    ucs_status_t status;
+
+    if (uct_flags & UCT_MD_MEM_FLAG_LOCK) {
+        reg_md_map |= context->reg_block_md_map[mem_type];
+    }
+
+    reg_md_map &= ~memh->md_map;
+    status = ucp_memh_init_uct_reg(context, memh, reg_md_map, uct_flags,
+                                   alloc_name);
```

Comment:
maybe pass "reg_md_map &  ~memh->md_map" to ucp_memh_init_uct_reg instead of modifying reg_md_map?